src/header.rs

```diff
@@ -202,167 +1398,69 @@ impl Default for NiftiHeader {
-        let len = self.descrip.len();
-        if len > 80 {
-            Err(NiftiError::IncorrectDescriptionLength(len))
-        } else {
-            if len < 80 {
-                self.descrip.extend((len..80).map(|_| 0));
-            }
-            Ok(())
-        }
-    }
+impl Default for Nifti2Header {
+    fn default() -> Nifti2Header {
+        Nifti2Header {
+            sizeof_hdr: 540,
+            magic: *MAGIC_CODE_NIP2,
+            datatype: 0,
+            bitpix: 0,
+            dim: [1, 0, 0, 0, 0, 0, 0, 0],
+            intent_p1: 0.,
+            intent_p2: 0.,
+            intent_p3: 0.,
+            pixdim: [1.; 8],
+            vox_offset: 544,
+            scl_slope: 0.,
+            scl_inter: 0.,
+            cal_max: 0.,
+            cal_min: 0.,
+            slice_duration: 0.,
+            toffset: 0.,
+            slice_start: 0,
+            slice_end: 0,
+            descrip: [0; 80],
+            aux_file: [0; 24],
+            qform_code: 1,
+            sform_code: 1,
+            quatern_b: 0.,
+            quatern_c: 0.,
+            quatern_d: 0.,
+            quatern_x: 0.,
+            quatern_y: 0.,
+            quatern_z: 0.,
+            srow_x: [1., 0., 0., 0.],
+            srow_y: [0., 1., 0., 0.],
+            srow_z: [0., 0., 1., 0.],
+            slice_code: 0,
+            xyzt_units: 0,
+            intent_code: 0,
+            intent_name: [0; 16],
+            dim_info: 0,
 
-    /// Safely set the `descrip` field using a buffer.
-    pub fn set_description<D>(&mut self, description: D) -> Result<()>
-    where
-        D: Into<Vec<u8>>,
-        D: Deref<Target = [u8]>,
-    {
-        let len = description.len();
-        match len.cmp(&80) {
-            std::cmp::Ordering::Less => {
-                let mut descrip = vec![0; 80];
-                descrip[..len].copy_from_slice(&description);
-                self.descrip = descrip;
-                Ok(())
-            }
-            std::cmp::Ordering::Equal => {
-                self.descrip = description.into();
-                Ok(())
-            }
-            _ => Err(NiftiError::IncorrectDescriptionLength(len)),
+            endianness: Endianness::native(),
         }
     }
+}
 
-    /// Safely set the `descrip` field using a  &str.
-    pub fn set_description_str<T>(&mut self, description: T) -> Result<()>
-    where
-        T: Into<String>,
-    {
-        self.set_description(description.into().as_bytes())
+impl Nifti1Header {
+    /// Place this `Nifti1Header` into a version-agnostic [`NiftiHeader`] enum.
+    pub fn into_nifti(self) -> NiftiHeader {
+        NiftiHeader::Nifti1Header(self)
     }
+}
 
-    /// Check whether `pixdim[0]` is either -1 or 1.
-    #[inline]
-    fn is_pixdim_0_valid(&self) -> bool {
-        (self.pixdim[0].abs() - 1.).abs() < 1e-11
+impl Nifti2Header {
+    /// Place this `Nifti2Header` into a version-agnostic [`NiftiHeader`] enum.
+    pub fn into_nifti(self) -> NiftiHeader {
```

Comment:
It would be idiomatic to also have implementations of `From<Nifti1Header>` and `From<Nifti2Header>` for `NiftiHeader`. This way, a conversion would be as simple as `header.into()`.